test/release/examples/primers/forallLoops.chpl

```diff
@@ -0,0 +1,305 @@
+writeln("outerAtomicVariable is: ", outerAtomicVariable.read());
+writeln();
+
+/*
+The task intents ``in``, ``const in``, ``ref``, ``const ref``,
+and ``reduce`` can be specified explicitly using a ``with`` clause.
+
+An ``in`` or ``const in`` intent creates a copy of the outer variable
+for each task. A ``ref`` or ``const ref`` makes the
+shadow variable an aliass for the outer variable.
+*/
+
+var outerRealVariable = 1.0;
+
+forall i in 1..n with (in outerIntVariable,
+                       ref outerRealVariable) {
+  outerIntVariable += 1; // a per-task copy, never accessed concurrently
+
+  if i == 1 then
+    outerRealVariable *= 2;  // beware of potential for data races
+}
+
+writeln("After a loop with explicit intents:");
+writeln("outerIntVariable is: ", outerIntVariable);
+writeln("outerRealVariable is: ", outerRealVariable);
+writeln();
+
+/*
+A reduce intent can be used to compute reductions.
+The values of each reduce-intent shadow variable at the end of its task
+is combined onto its outer variable according to the specified reduction
+operation. Within loop body, the shadow variable represents the
+accumulation state produced by this task so far, starting from
+the reduction identity value at task startup. Values can be
+combined onto this accumulation state using the reduction-specific
+operation or the ``reduce=`` operator.
+*/
+
+var outerMaxVariable = 0;
+// outerIntVariable's value before the loop will be included in the sum
+
+forall i in 1..n with (+ reduce outerIntVariable,
+                       max reduce outerMaxVariable) {
+  outerIntVariable += i;
+  if i % 2 == 0 then
+    outerMaxVariable reduce= i;
+
+  // The loop body can contain other code
+  // regardless of reduce-related operations.
+}
+
+writeln("After a loop with reduce intents:");
+writeln("outerIntVariable = ", outerIntVariable);
+writeln("outerMaxVariable = ", outerMaxVariable);
+writeln();
+
+/*
+A with-clause can be used in a similar fashion with any flavor
+of forall loop.
+
+Task-Private Variables
+----------------------
+
+A task-private variable is similar to an in-intent or ref-intent
+shadow variable in that it is initialized at the beginning of its
+task, shared by all loop iterations executed by the task, and
+deallocated at the end of the task. However, a task-private
+variable is initialized without regard to any outer variable.
+
+A task-private variable is introduce using a with-clause
+in a way similar to a regular ``var``, ``const``, ``ref``,
+or ``const ref`` variable. A ``var`` or ``const`` variable
+must provide its type or initializing expression or both.
+As with a regular variable, it will be initialized
+to the default value of its type the initializing expression
+is not given. A ``ref`` or ``const ref`` variable must
+have the initializing expression and cannot declare its type.
+
+A ``var`` task-private variable could be used, for example,
+as a per-task scratch space that is never accessed concurrently.
```

Comment:
I'd start this section with an example of its use *before* going into the specifications.  It helps to motivate the reader.  Something like this paragraph, but rephrased so it isn't dependent on previous paragraphs